tests/features/payment-gateway-suggestions/payment-gateway-controller.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * Test the class that extends the payment gateway REST response.
+ *
+ * @package WooCommerce\Admin\Tests\PaymentGatewaySuggestions
+ */
+
+use Automattic\WooCommerce\Admin\Features\RemotePaymentMethods\PaymentGatewaysController;
+
+/**
+ * class WC_Tests_PaymentGatewaySuggestions_PaymentGatewaysController
+ */
+class WC_Tests_PaymentGatewaySuggestions_PaymentGatewaysController extends WC_REST_Unit_Test_Case {
+	/**
+	 * Endpoints.
+	 *
+	 * @var string
+	 */
+	protected $endpoint = '/wc/v3/payment_gateways';
+
+	/**
+	 * Set up.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->user = $this->factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+		wp_set_current_user( $this->user );
+
+		add_filter(
+			'woocommerce_payment_gateways',
+			function( $gateways ) {
+				$gateways[] = 'WC_Mock_Payment_Gateway';
+
+				return $gateways;
+			}
+		);
+		WC()->payment_gateways()->init();
+
+		$this->gateway = new WC_Mock_Payment_Gateway();
+	}
+
+	/**
+	 * Get mock gateway.
+	 */
+	public function get_mock_gateway_response() {
+		$request  = new WP_REST_Request( 'GET', $this->endpoint . '/mock' );
+		$response = $this->server->dispatch( $request );
+		return $response->get_data();
+	}
+
+	/**
+	 * Tests that needs_setup is initially false.
+	 */
+	public function test_payment_gateway_needs_setup() {
+		$response = $this->get_mock_gateway_response();
+		$this->assertFalse( $response['needs_setup'] );
+	}
+
+	/**
+	 * Tests that the gateway no longer needs setup.
+	 */
+	public function test_payment_gateway_setup() {
+		$this->gateway->update_option( 'api_key', '123' );
+		$response = $this->get_mock_gateway_response();
+		$this->assertTrue( $response['needs_setup'] );
+	}
+
+	/**
+	 * Tests the gateways post install script handles and returned script dependencies.
+	 */
+	public function test_post_install_scripts() {
+		$response = $this->get_mock_gateway_response();
+		$this->assertCount( 1, $response['post_install_scripts'] );
```

Comment:
It might also be worth testing a gateway that doesn't contain the `get_post_install_script_handles` method, same goes for `get_connection_url`, `get_setup_help_text`, `get_required_settings_keys`